pkg/satellite/stream/stream.go

```diff
@@ -181,7 +182,36 @@ func (ss *satelliteStream) ackReceivedTelemetry(telemetryMessageAckId string) {
 	}
 }
 
+func (ss *satelliteStream) performAutoClose() {
+	log.Printf("Stream auto-close conditions met - exiting")
+	if ss.showStats {
+		metrics.logReport()
+	}
+	close(ss.closeTickerChan)
+	ss.Close()
+	os.Exit(0)
+}
+
 func (ss *satelliteStream) recvLoop() {
+	streamEndDetected := false
+
+	if ss.enableAutoClose {
+		ticker := time.NewTicker(1 * time.Second)
+		go func() {
+			for {
+				select {
+				case <-ticker.C:
+					if streamEndDetected {
+						ss.performAutoClose()
+					}
+				case <-ss.closeTickerChan:
```

Comment:
`ss.closeTickerChan` is being read from but never written to. What purpose does it serve?